examples/pyomobook/test_book_examples.py

```diff
@@ -42,6 +42,11 @@ def gurobi_fully_licensed():
 if not param_available:
     raise unittest.SkipTest('Parameterized is not available.')
 
+# Needed for testing (switches the matplotlib backend):
+from pyomo.common.dependencies import matplotlib_available
+
+bool(matplotlib_available)
```

Comment:
Shouldn't this already be a `bool`?

Reply:
No: it starts off as a `DeferredImportIndicator` object and only becomes a `bool` when it is cast (or when an attribute on the deferred module is accessed).  See https://pyomo.readthedocs.io/en/latest/library_reference/common/dependencies.html#pyomo.common.dependencies.attempt_import